lightning/src/ln/functional_test_utils.rs

```diff
@@ -1203,6 +1225,24 @@ macro_rules! expect_pending_htlcs_forwardable_ignore {
 	}}
 }
 
+#[macro_export]
+macro_rules! expect_warning_due_wrong_channel_reestablish {
+	($node: expr) => {
+		for msg in $node.node.get_and_clear_pending_msg_events() {
+		if let MessageSendEvent::HandleError { ref action, .. } = msg {
+			match action {
+				&ErrorAction::SendWarningMessage { ref msg, .. } => {
+					assert!(msg.data.contains(&"Peer attempted to reestablish channel with a very old local commitment transaction".to_owned()));
```

Comment:
Instead of a new macro, why not just extend `check_warn_msg` with the expected message? If not just inline this - I don't think we need to macro-ize this given we're unlikely to need to check this exact error string again.

Reply:
The only reason that I created a new macro is that I was not able to get working the contains method over an `expr` inside a macro.

The solution can be to simplify the error message to static `str`, where we can call over an `assert_eq!` assert.

What do you think? I'm missing something?